### PR TITLE
Add post-purchase motivation survey links for supporter plus

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
## Why are you doing this?
Post-purchase surveys live for supporter-plus (ONLY) to build out a portfolio of supporter and subscriber motivations, spanning both digital and print products.These surveys are built using Formstack links.

## What are you doing in this PR?

- Supporter sees survey element on post-purchase ‘thank you’ screen (location?)
- Depending on the type of purchase, the relevant survey link is used.

The change this time is that we’d like to link to 3 different surveys, conditional on the type of purchase:

1. [Recurring, below ‘extras’ threshold](https://guardiannewsandmedia.formstack.com/forms/guardian_supporter_below)
2. [Recurring, above ‘extras’ threshold](https://guardiannewsandmedia.formstack.com/forms/guardian_supporter_above)
3. [Single contributions](https://guardiannewsandmedia.formstack.com/forms/guardian_supporter_single)

## Screenshot
Screenshot 2023-06-23 at 15 29 54

[**Trello Card**](https://trello.com/c/Mm7a1qeP/1268-add-post-purchase-motivation-survey-links-for-supporter-plus)

## Is this an AB test?
- [ ] Yes
- [x] No

## Accessibility test checklist
 - [ ] [Tested with screen reader](https://webaim.org/articles/voiceover/)
 - [ ] [Navigable with keyboard](https://www.accessibility-developer-guide.com/knowledge/keyboard-only/browsing-websites/)
 - [ ] [Colour contrast passed](https://www.whocanuse.com/)

### CODE REVIEW

**Comment by paul-daniel-dempsey** [comment]
[Add post-purchase motivation survey links for supporter plus](https://trello.com/c/Mm7a1qeP/1268-add-post-purchase-motivation-survey-links-for-supporter-plus)

File: support-frontend/assets/pages/supporter-plus-thank-you/supporterPlusThankYou.tsx [hunk 1]
```diff
@@ -187,7 +190,7 @@ export function SupporterPlusThankYou(): JSX.Element {
 			contributionType === 'ONE_OFF' && email.length > 0,
 			'supportReminder',
 		),
-		// 'feedback', // <-- Temporarily disable supporter plus thank you page survey
+		...maybeThankYouModule(email.length > 0, 'feedback'),
```
Comment (undergroundquizscene) [inline support-frontend/assets/pages/supporter-plus-thank-you/supporterPlusThankYou.tsx:193]: Do we sometimes not have an email address at this point?

**Review by undergroundquizscene — APPROVED** [review]

File: support-frontend/assets/pages/supporter-plus-thank-you/supporterPlusThankYou.tsx (hunk 1 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/supporter-plus-thank-you/supporterPlusThankYou.tsx:193]: I don't know of a production scenario where you would get through the checkout to the thank-you without providing an email address (don't you need this for all checkouts?). Is there another route through I have not spotted?

Comment (undergroundquizscene) [reply support-frontend/assets/pages/supporter-plus-thank-you/supporterPlusThankYou.tsx:193]: Not that I know of, just wondering why you have the length check on the email address? (Happy to leave it in, but just not sure if it’s necessary)

**Comment by worldofpablo** [comment]
Tested both single and recurring (above and below the benefits threshold) and each time saw the correct survey links on the thank you page. Happy days.

File: support-frontend/assets/pages/supporter-plus-thank-you/supporterPlusThankYou.tsx (hunk 1 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/supporter-plus-thank-you/supporterPlusThankYou.tsx:193]: Paul Heron made a good point when checking, it also stops anyone spamming the thankyou url direct . Thus stopping  being able to post multiple survey's without an associated checkout.

**Pull request merged**